src/plugins/intel_cpu/src/transformation_pipeline.cpp

```diff
@@ -571,10 +570,34 @@ void Transformations::MainSnippets(void) {
     if (!isMHASupported) {
         snippetsManager.get_pass_config()->disable<ngraph::snippets::pass::TokenizeMHASnippets>();
     }
+
+    auto is_supported_matmul = [](const std::shared_ptr<const ov::op::v0::MatMul>& matmul) {
+        if (!matmul)
+            return false;
+        if (matmul->get_input_element_type(1) == ov::element::i8)
+            return dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_vnni);
+        if (matmul->get_input_element_type(0) == ov::element::bf16 &&
+            matmul->get_input_element_type(1) == ov::element::bf16)
+            return dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16);
+        return true;
+    };
+
     if (snippetsMode != Config::SnippetsMode::IgnoreCallback) {
         snippetsManager.get_pass_config()->set_callback<ngraph::snippets::pass::TokenizeMHASnippets>(
-                [](const std::shared_ptr<const ov::Node>& n) -> bool {
-                    const auto pshape = n->get_output_partial_shape(0);
+                [this, is_supported_matmul](const std::shared_ptr<const ov::Node>& n) -> bool {
+                    if (this->enableLpt) {
```

Comment:
Minor: you don't have to specify `this` explicitly for such cases (unless there is another variable with the same name that's also captured)